platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/TabStrip.kt

```diff
@@ -47,38 +59,124 @@ import org.jetbrains.jewel.ui.component.styling.TabStyle
  * @see javax.swing.JTabbedPane
  * @see com.intellij.ui.components.JBTabbedPane
  */
 @Composable
-public fun TabStrip(tabs: List<TabData>, style: TabStyle, modifier: Modifier = Modifier, enabled: Boolean = true) {
+public fun TabStrip(
+    tabs: List<TabData>,
+    style: TabStyle,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+) {
     var tabStripState: TabStripState by remember { mutableStateOf(TabStripState.of(enabled = true)) }
+    val selectedIndex = remember(tabs) { tabs.indexOfFirst { it.selected } }
+    val tabCount = tabs.size
 
     remember(enabled) { tabStripState = tabStripState.copy(enabled) }
 
     val scrollState = rememberScrollState()
+
     Box(
-        modifier.focusable(true, remember { MutableInteractionSource() }).onHover {
-            tabStripState = tabStripState.copy(hovered = it)
-        }
+        modifier
+            .onFocusEvent { tabStripState = tabStripState.copy(focused = it.isFocused) }
+            .onHover { tabStripState = tabStripState.copy(hovered = it) }
+            .onPreviewKeyEvent { event ->
+                if (!enabled || tabs.isEmpty()) return@onPreviewKeyEvent false
+
+                when (event.type) {
+                    KeyEventType.KeyDown ->
+                        when (event.key) {
+                            Key.DirectionLeft -> {
+                                tabs[if (selectedIndex > 0) (selectedIndex - 1) else tabs.lastIndex].onClick()
+                                true
+                            }
+
+                            Key.DirectionRight -> {
+                                tabs[(selectedIndex + 1) % tabCount].onClick()
+                                true
+                            }
+
+                            else -> false
+                        }
+
+                    KeyEventType.KeyUp ->
+                        when (event.key) {
+                            Key.MoveHome -> {
+                                tabs.first().onClick()
+                                true
+                            }
+
+                            Key.MoveEnd -> {
+                                tabs.last().onClick()
+                                true
+                            }
+
+                            Key.Delete -> {
+                                if (tabs[selectedIndex].closable) {
+                                    tabs[selectedIndex].onClose()
+                                    true
+                                } else {
+                                    false
+                                }
+                            }
+
+                            else -> false
+                        }
+
+                    else -> false
+                }
+            }
+            .focusable(enabled, interactionSource)
+            .hoverable(interactionSource, enabled)
```

Comment:
Don't these two duplicate what `onFocusEvent` and `onHover` do? If they're only here to update the interactionSource, I'd try to unify the two functionalities